src/features/device-state/routes/state.ts

```diff
@@ -25,6 +26,163 @@ export const varListInsert = (varList: EnvVarList, obj: Dictionary<string>) => {
 	});
 };
 
+export type App = {
+	name: string;
+	commit: string;
+	releaseId: number;
+	releaseVersion: string;
+	appId?: string;
+	uuid?: string;
+	services: {
+		[id: number]: AppService;
+	};
+	volumes: any;
+	networks: any;
+};
+
+export type AppService = {
+	imageId: number;
+	serviceName: string;
+	image: string;
+	running: boolean;
+	environment: Dictionary<string>;
+	labels: Dictionary<string>;
+	contract?: string;
+};
+
+export type LocalState = {
+	name: string;
+	config: Dictionary<string>;
+	apps: Dictionary<Partial<App>>;
+};
+
+async function buildAppFromRelease(
+	device: AnyObject,
+	application: AnyObject,
+	release: AnyObject,
+	config: Dictionary<string>,
+): Promise<App> {
+	let composition: any = {};
+	const services: App['services'] = {};
+
+	// Parse the composition to forward values to the device
+	if (_.isObject(release.composition)) {
+		composition = release.composition;
+	} else {
+		try {
+			composition = JSON.parse(release.composition);
+		} catch (e) {
+			composition = {};
+		}
+	}
+
+	(release.contains__image as AnyObject[]).forEach((ipr) => {
+		// extract the per-image information
+		const image = ipr.image[0];
+
+		const serviceInstall = (() => {
+			const si = serviceInstallFromImage(device, image);
+
+			if (si != null) {
+				return si;
+			}
+
+			if (image.is_a_build_of__service[0] == null) {
+				throw new Error(
+					`Could not find service install for device: '${
+						device.uuid
+					}', image: '${image?.id}', service: '${JSON.stringify(
+						image?.is_a_build_of__service,
+					)}', service installs: '${JSON.stringify(device.service_install)}'`,
+				);
+			}
+
+			/**
+			 * Return a fake service install as this is what we will use for MultiApp
```

Comment:
Why do we have fake service installs rather than real ones?

Reply:
I think this is a vestigial piece of the original implementation for system apps like ingrid -- before true multiapp, we weren't able to properly associate a service_install with a device (maybe?)

Reply:
I need some guidance on this bit of code... it seems that it's purpose is to guarantee that `serviceInstall` always defined so that we can proceed with the rest of the code.  The "fake service" bit only runs if a real service install was not retrieved.   Do I remove it?  

This is especially confusing given that `utils.ts/serviceInstallFromImage` has changed significantly.  

We went from this:
```
	if (typeof image.is_a_build_of__service === 'object') {
		id = image.is_a_build_of__service.__id;
	} else {
		id = image.is_a_build_of__service;
	}
```
to this
```
	if (Array.isArray(image.is_a_build_of__service)) {
		id = image.is_a_build_of__service[0].id;
	} else {
		id = image.is_a_build_of__service.__id;
	}
```
implying that the nature of `is_a_build_of__service` has changed.   Also, it now looks in the current code like an exception is possible if `is_a_build_of__service` is undefined.  
@xginn8 gonna need  your advice on how to proceed.

Reply:
i think service installs are created typically for all required services x devices, but only in the case of system apps could they not exist since it was kind of shoehorned into the model. i think it's best to remove, but i'm not 100% confident on that either.

Reply:
After more investigation this is necessary for the supervisor to appear to be an app.  The only alternative would be to create a service install when we PATCH to link the supervisor.

Reply:
I think the correct answer is indeed to add those service installs @toochevere since it moves towards the future where it's a real app rather than hacking around the fact that it's not quite a true app yet